src/tests/Microsoft.PowerFx.Interpreter.Tests/LanguageServiceProtocol/SemanticTokensEncoderTests.cs

```diff
@@ -197,12 +200,46 @@ public void TestMultilineTokensAreEncodedCorrectly(string expression)
             var eol = ChooseEol(expression);
 
             // Act
-            var encodedTokens = SemanticTokensEncoder.EncodeTokens(tokens, expression, eol);
+            var encodedTokens = SemanticTokensEncoder.EncodeTokens(tokens, expression, eol, null);
 
             // Assert
             AssertEncodedTokens(encodedTokens, tokens, expression, eol, true);
         }
 
+        [Theory]
+        [InlineData("9 + 9", 0)]
+        [InlineData("Max(10, 20, 30)", 0)]
+        [InlineData("Color.AliceBlue", 0)]
+        [InlineData("9 + 9\n Max(1, 3, 19); \n Color.AliceBlue", 0)]
+        [InlineData("/*jj*/\n\r\n\r\n//yes", 0)]
+        [InlineData("Label2.Text", 1, new string[] { "Label2,0,0,0,6" })]
+        [InlineData("NestedLabel1", 1, new string[] { "NestedLabel1,0,0,0,12" })]
+        [InlineData("Label2.Text;\nNestedLabel1", 2, new string[] { "Label2,0,0,0,6", "NestedLabel1,1,0,1,12" })]
+        [InlineData("Label2.Text;Label2.Text", 1, new string[] { "Label2,0,0,0,6,0,12,0,18" })]
+        public void TestControlTokensAreComputedCorrectlyWhenPresent(string expression, int expectedNumberOfControlTokens, string[] expectedControlTokenIndicesArray = null)
```

Comment:
Can we add one more test case here? gallery.Selected.Label2 where both gallery and Label2 are controls. Look into the MockSymbolTable in TOkenizatonTests.cs to mock this.